### PR TITLE
Add path to NeuralNet.h when USE_TENSORRT=1

### PR DESCRIPTION
**Pull request by rjknight**
Fix the compile fail when USE_TENSORRT=1

../caffe2/opt/backend_cutting.h:5:10: fatal error:
      nomnigraph/Representations/NeuralNet.h: No such file or directory

see issue: https://github.com/pytorch/pytorch/issues/31916

### CODE REVIEW

File: cmake/Dependencies.cmake [hunk 1]
```diff
@@ -1265,6 +1265,7 @@ if(CAFFE2_CMAKE_BUILDING_WITH_MAIN_REPO)
     set(CMAKE_CUDA_COMPILER ${CUDA_NVCC_EXECUTABLE})
     add_onnx_tensorrt_subdir()
     include_directories("${CMAKE_CURRENT_LIST_DIR}/../third_party/onnx-tensorrt")
+    include_directories("${CMAKE_SOURCE_DIR}/caffe2/core/nomnigraph/include")
```
Comment (ZolotukhinM) [inline cmake/Dependencies.cmake:1268]: I'm not sure `cmake/Dependencies.cmake` is the right place for it. Shouldn't it be set somewhere closer to where it's used, like `caffe2/contrib/tensorrt/CMakeLists.txt`?

**Review by malfet — CHANGES_REQUESTED** [review]
Hmm, it looks like all one has to do to solve the issue, is to move
`add_subdirectory(core/nomnigraph)` before
https://github.com/pytorch/pytorch/blob/64594d83336c28ea5f07b8572c9bcb1f0e7abde4/caffe2/CMakeLists.txt#L77

**Comment by rjknight** [comment]
Modifying the placement of the add_subdirectory didn't seem to make a difference, after adjusting the code I still see the failure. 
I enabled the debug code at the top fo caffe2/CMakeLists.txt and can see the needed include directory is only in the Caffe2_CPU_INCLUDE variable
```
2292 -- CPU include: 
2293 --   $SRC_DIR/build/caffe2/aten/src/TH
2294 --   $SRC_DIR/aten/src/TH
2295 --   $SRC_DIR/aten/src
2296 --   $SRC_DIR/build/caffe2/aten/src
2297 --   $SRC_DIR/build/aten/src
2298 --   $SRC_DIR/aten/src
2299 --   $SRC_DIR/aten/../third_party/catch/single_include
2300 --   $SRC_DIR/aten/src/ATen/..
2301 --   $SRC_DIR/build/caffe2/aten/src/ATen
2302 --   $SRC_DIR/third_party/miniz-2.0.8
2303 --   $SRC_DIR/caffe2/core/nomnigraph/include
2304 -- GPU include: 
2305 --   $PREFIX/include
2306 --   $SRC_DIR/build/caffe2/aten/src/TH
2307 --   $SRC_DIR/aten/src/TH
2308 --   $SRC_DIR/build/caffe2/aten/src/THC
2309 --   $SRC_DIR/aten/src/THC
2310 --   $SRC_DIR/aten/src/THCUNN
2311 --   $SRC_DIR/aten/src/ATen/cuda
2312 --   $SRC_DIR/build/caffe2/aten/src/TH
2313 --   $SRC_DIR/aten/src/TH
2314 --   $SRC_DIR/aten/src
2315 --   $SRC_DIR/build/caffe2/aten/src
2316 --   $SRC_DIR/build/aten/src
2317 --   $SRC_DIR/aten/src
2318 --   $SRC_DIR/aten/../third_party/catch/single_include
2319 --   $SRC_DIR/aten/src/ATen/..
2320 --   $SRC_DIR/build/caffe2/aten/src/ATen
```
I found that Caffe2_CPU_INCLUDE is currently getting set here - https://github.com/pytorch/pytorch/blob/64594d83336c28ea5f07b8572c9bcb1f0e7abde4/caffe2/core/nomnigraph/CMakeLists.txt#L7
seems like I could just add an additional case for adding the path to Caffe2_GPU_INCLUDE but maybe setting it in the caffe2/contrib/tensorrt/CMakeLists.txt as suggested makes more sense.  I'm leaning that way, any thoughts or preference as to the location?

**Comment by malfet** [comment]
@rjknight sorry for the late reply. Will look into why it is not working as expected today.

**Comment by malfet** [comment]
Sorry for the late reply again, how about something like https://github.com/pytorch/pytorch/pull/37056

**Comment by rjknight** [comment]
> Sorry for the late reply again, how about something like #37056

Looks good, I'll give it a try.  Thanks!

**Comment by rjknight** [comment]
Closing in favor of https://github.com/pytorch/pytorch/pull/37056.  

Thanks @malfet

**Pull request closed without merging**